modules/network/src/main/java/org/apache/ignite/internal/network/recovery/RecoveryServerHandshakeManager.java

```diff
@@ -177,12 +182,20 @@ public void onMessage(NetworkMessage message) {
         if (message instanceof HandshakeRejectedMessage) {
             HandshakeRejectedMessage msg = (HandshakeRejectedMessage) message;
 
-            LOG.warn("Handshake rejected by client: {}", msg.reason());
+            boolean ignorable = stopping.get() || !msg.critical();
```

Comment:
Why do we check the stopping flag here?

Reply:
If a node are stopping, then the situation 'I was not let by another node to do a handshake' is expected and should neither cause anything (serious) be logged nor fail the node

Reply:
Agree